pkg/operator/ceph/object/admin.go

```diff
@@ -482,16 +482,17 @@ func GetAdminOPSUserCredentials(objContext *Context, spec *cephv1.ObjectStoreSpe
 		forceUserCreation = true
 	}
 
-	user, rgwerr, err := CreateUser(objContext, userConfig, forceUserCreation)
+	user, rgwerr, err := GetUser(objContext, userConfig.UserID)
 	if err != nil {
-		if rgwerr == ErrorCodeFileExists {
-			user, _, err = GetUser(objContext, userConfig.UserID)
-			if err != nil {
-				return "", "", errors.Wrapf(err, "failed to get details from ceph object user %q for object store %q", userConfig.UserID, objContext.Name)
-			}
-		} else {
+		if rgwerr != RGWErrorNotFound {
+			// any other error is unexpected, but we can attempt to create the user anyway. log the err to help debug corner cases
+			logger.Infof("failed to get details from ceph object user %q for object store %q. code=%d. %v", userConfig.UserID, objContext.Name, rgwerr, err)
+		}
```

Comment:
When reordering these commands, I am worried that we might not always get RGWErrorNotFound (code: 2) when the admin ops user doesn't exist. It seemed best to me to log any other errors with info level so we can see them in logs easily, and then attempt to create the admin ops user.

I'm still a little hesitant about this aspect of the change.

Also, this appears to have some good discussion here: https://github.com/rook/rook/pull/8772

I'm not sure why Seb didn't follow up to change this after Jiffin's comment ~4 years ago.

Reply:
At least one object suite passed, so I think we can have a pretty high confidence that this doesn't break anything at least